networkx/algorithms/dag.py

```diff
@@ -572,6 +572,7 @@ def all_topological_sorts(G):
             break
 
 
+@not_implemented_for("undirected")
```

Comment:
Not sure if this will require a deprecation. In any case, we should also update the docstring and get rid of the `if not G.is_directed()` at the start of the function.

Reply:
No deprecation required if it already raises when not directed. The decorator doesn't change behavior, only where the checking/raising occurs.  So the checking and raising inside the funcgtion should be removed if the decorator is included.

Reply:
Even when the type of exception has changed from `NetworkXError` to `NetworkXNotImplemented`?

Reply:
Ahh -- Thanks for noticing that the type of the exception has changed!!

Luckily `NetworkXNotImplemented` is a subclass of `NetworkXError` (I hope). So the original check should still work.  Does it?

Reply:
The original test failed with the change and it is updated.

Reply:
It isn't, unfortunately; `NetworkXNotImplemented` is a subclass of `NetworkXException`:
```python
>>> issubclass(nx.NetworkXNotImplemented, nx.NetworkXError)
False
>>> issubclass(nx.NetworkXNotImplemented, nx.NetworkXException)
True
```

Reply:
Should we revert the change to avoid deprecations or should we use `not_implemented_for` to stay consistent with the rest of the code? What do you think?

Reply:
If we decide to go with `NetworkXNotImplemented` (which makes the most sense to me and is what nearly all other functions in the dag.py file do), the "Raises" section in the docstring also needs to be changed to reflect that.

Reply:
Yes, docstrings should be updated. We may also need to set up the function for deprecation since it raises a different exception (not sure).

We haven't decided whether to apply the change or not yet.

Reply:
Let's go with `NetworkXNotImplemented`, don't worry about deprecation and change the Raises section of the docs. At least that's what I would prefer.